gpustat.py

```diff
@@ -262,22 +263,22 @@ def running_processes():
                                   })
             process_entries.append(process_entry)
 
-        pid_map = {int(e['pid']) : None for e in process_entries
-                   if not 'Not Supported' in e['pid']}
-
-        # 2. map pid to username, etc.
-        if pid_map:
-            pid_output = execute_process('ps -o {} -p {}'.format(
-                'pid,user:16,comm',
-                ','.join(map(str, pid_map.keys()))
-            ))
-            for line in pid_output.split('\n'):
-                if (not line) or 'PID' in line: continue
-                pid, user, comm = line.split()[:3]
-                pid_map[int(pid)] = {
-                    'user' : user,
-                    'comm' : comm
-                }
+        pid_map = {int(e['pid']) : None for e in process_entries if not 'Not Supported' in e['pid']}
+        if os.environ.get('DOCKER_CONTAINER') is  None:
```

Comment:
The problem is nvidia-smi returns process ids running on the host.  The container does not necessarily have access to those pids.  When the docker program attempts to introspect the host pids the program exits on bad return. 

I'll paste some debug when I get back into work next week.